src/librustc_builtin_macros/asm.rs

```diff
@@ -264,27 +270,72 @@ fn parse_args<'a>(
     Ok(args)
 }
 
+fn warn_duplicate_option<'a>(p: &mut Parser<'a>, span: Span) {
+    let mut warn = if let Ok(snippet) = p.sess.source_map().span_to_snippet(span) {
+        p.sess
+            .span_diagnostic
+            .struct_span_warn(span, &format!("the `{}` option was already provided", snippet))
+    } else {
+        p.sess.span_diagnostic.struct_span_warn(span, "this option was already provided")
+    };
+    warn.span_suggestion(
+        span,
+        "remove this option",
+        String::new(),
+        Applicability::MachineApplicable,
+    );
```

Comment:
The suggestion isn't actually machine applicable since you may end up with two stray commas next to each other, which would be a syntax error.

Reply:
`Applicability::Unspecified`?

Reply:
I'm tending towards dropping the suggestion entirely. I think the error message is clear enough on its own.

Reply:
I like that the suggestion tells you exactly how you can remedy the problem though.

Reply:
That way the user knows that they can fix the problem easily, and it's not because of some other cause.

Reply:
A `span_help` would probably be more appropriate in this case.

Reply:
Okay, sounds good

Reply:
Weirdly, after I switched to using `span_help`, the help message is in a separate diagnostics message.

**Before:**

```
error: the `att_syntax` option was already provided
  --> rust-issue-73193.rs:10:21
   |
10 |             options(att_syntax),
   |                     ^^^^^^^^^^ help: remove this option
```

**After:**

```
error: the `att_syntax` option was already provided
  --> rust-issue-73193.rs:10:21
   |
10 |             options(att_syntax),
   |                     ^^^^^^^^^^
   |
help: remove this option
  --> rust-issue-73193.rs:10:21
   |
10 |             options(att_syntax),
   |                     ^^^^^^^^^^
```

How do I get it back to one message?

Reply:
Use `span_label`.

Reply:
After reviewing the suggestion API in detail, I think your original plan of using `suggestion` with `Applicability::Unspecified` is probably the correct choice. I can see other places in rustc that do something similar.

Reply:
Will that then give the user the option to apply the suggestion if they're in an IDE?

Reply:
I think `span_suggestion_short` might be the best fit